### PR TITLE
[pkg/winperfcounters] Retry collection when rolloever error is detected

### LINKED ISSUE #11445: [receiver/windowsperfcounters] Flaky Test - Test_WindowsPerfCounterScraper
Observed here: https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/7008709942?check_suite_focus=true#step:7:2750

```
=== RUN   Test_WindowsPerfCounterScraper/Standard
    windowsperfcounters_scraper_test.go:184: 
        	Error Trace:	windowsperfcounters_scraper_test.go:184
        	Error:      	Received unexpected error:
        	            	A counter with a negative denominator value was detected.
        	Test:       	Test_WindowsPerfCounterScraper/Standard
=== RUN   Test_WindowsPerfCounterScraper/SumMetric
=== RUN   Test_WindowsPerfCounterScraper/NoMetricDefinition
=== RUN   Test_WindowsPerfCounterScraper/InvalidCounter
--- FAIL: Test_WindowsPerfCounterScraper (0.25s)
    --- FAIL: Test_WindowsPerfCounterScraper/Standard (0.25s)
```

### PR DESCRIPTION
**Pull request by djaglowski**
Based on documentation, I believe this resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11445. However, I'm not sure how to test it as it occurs very infrequently.

### CODE REVIEW

**Comment by dmitryax** [comment]
@djaglowski can you please rebase?

File: unreleased/winperf-retry-rollover.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Retry counter when rollover error is detected
+
+# One or more tracking issues related to the change
+issues: [14343]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (dmitryax) [inline unreleased/winperf-retry-rollover.yaml:16]: This file must be redundant?

File: pkg/winperfcounters/watcher.go [hunk 2]
```diff
@@ -104,9 +105,19 @@ func (pc *perfCounter) Path() string {
...
+		if !ok || pdhErr.ErrorCode != win_perf_counters.PDH_CALC_NEGATIVE_DENOMINATOR {
+			return nil, fmt.Errorf("failed to collect data for performance counter '%s': %w", pc.path, err)
+		}
+
+		// A counter rolled over, so the value is invalid
+		// See https://support.microfocus.com/kb/doc.php?id=7010545
+		// Wait one second and retry once
+		time.Sleep(time.Second)
+		if retryErr := pc.query.CollectData(); retryErr != nil {
+			return nil, retryErr
```
Comment (dmitryax) [inline pkg/winperfcounters/watcher.go:119]: Should we wrap the error here as well?

File: unreleased/winperf-retry-rollover.yaml (hunk 1 above)
Comment (djaglowski) [reply unreleased/winperf-retry-rollover.yaml:16]: Good catch

File: unreleased/add-azureblob-receiver.yaml [hunk 3]
```diff
@@ -1,16 +0,0 @@
...
-# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
-note: Add a new component `azureblobreciver`
-
-# One or more tracking issues related to the change
-issues: [8834]
-
-# (Optional) One or more lines of additional information to render under the primary note.
-# These lines will be padded with 2 spaces and then inserted directly into the document.
-# Use pipe (|) for multiline entries.
-subtext: Add a new component `azureblobreceiver` that reads logs and traces from Azure Blob Storage
```
Comment (dmitryax) [inline unreleased/add-azureblob-receiver.yaml:16]: Are they really duplicates or maybe missed? Anyway, it's probably a good candidate for another PR

Comment (djaglowski) [reply unreleased/add-azureblob-receiver.yaml:16]: Sorry, apparently another PR leaked into this one. Thanks for catching this.

**Review by dmitryax — APPROVED** [review]

**Comment by djaglowski** [comment]
I realized also that #12956 had been merged since this PR was opened, so this also reverts that change.

**Pull request merged**